src/library/scala/collection/ArrayOps.scala

```diff
@@ -1637,4 +1637,12 @@ final class ArrayOps[A](private val xs: Array[A]) extends AnyVal {
     *  @return `true` if this array has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = mutable.ArraySeq.make(xs).endsWith(that)
+
+  /** Ensure we are providing an array of the correct element type.
+   */
+  private def narrowly(ys: Array[_]): Array[A] =
```

Comment:
can we make this method accept `mutable.ArraySeq` instead since all callers seem to use it:

```scala
private def narrowArray(ys: mutable.ArraySeq[A]): Array[A] = ...
```
this makes it somewhat safer, IMO, in that the contract on the input is strengthened.